### DISCUSSION TITLE
Do basic Linux GitHub-hosted runners really feature two cores?

### DISCUSSION THREAD

**Discussion by pguyot** [community/community#37306] [Actions]
I do have a test that relies on the host to actually feature 2 or more cores. This tests consistently fails on GitHub-hosted Linux runners but succeeds on macOS runners. It succeeds locally using docker / VirtualBox with 2 or 3 virtual CPUs.

To investigate, I did write a small C code to assess the speed of two POSIX threads vs 1 and the results are a little bit surprising.

https://gist.github.com/pguyot/79afbf4091ac5bcb51e3ac1f8732042d

The output on Ubuntu 20.04 is the following:
https://github.com/pguyot/AtomVM/actions/runs/3326806636/jobs/5500898970#step:10:19

```
one thread: 295 us, two threads: 564 us
one thread: 272 us, two threads: 535 us
one thread: 274 us, two threads: 537 us
one thread: 290 us, two threads: 536 us
one thread: 255 us, two threads: 495 us
one thread: 253 us, two threads: 495 us
one thread: 249 us, two threads: 490 us
one thread: 268 us, two threads: 538 us
one thread: 266 us, two threads: 507 us
one thread: 249 us, two threads: 490 us
Check SMP failed, disabling prime_smp test
```

While on macOS runner it looks like this:
https://github.com/pguyot/AtomVM/actions/runs/3326806624/jobs/5500899704#step:7:10

```
one thread: 1939 us, two threads: 2003 us
one thread: 1985 us, two threads: 1990 us
one thread: 1979 us, two threads: 1982 us
one thread: 1946 us, two threads: 1984 us
one thread: 1928 us, two threads: 1987 us
one thread: 1931 us, two threads: 2015 us
one thread: 1990 us, two threads: 1996 us
one thread: 1983 us, two threads: 1968 us
one thread: 1967 us, two threads: 1973 us
one thread: 1985 us, two threads: 1992 us
```

Is there something faulty in my logic or are the GitHub-hosted runners not really featuring two cores?

**Comment by ossanna16** [comment]
Hi there @pguyot and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)
